beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -939,20 +945,34 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
 
         let connected_peer_count = self.network_globals.connected_peers();
         if connected_peer_count > self.target_peers {
-            //remove excess peers with the worst scores, but keep subnet peers
-            for (peer_id, _) in self
+            // Remove excess peers with the worst scores, but keep subnet peers.
```

Comment:
After some bisect digging, this code was originally added in #1383. By that time we were not handling the intermediate "Disconnecting" peers. This was later added in #1796. I think this additional logic to account for is no longer needed and as noted by all, it hinders readability. 
If a peer got to this list (`worst_connected_peers`) it means it is really connected.

I'd prefer if you try to remove the "is healthy logic" and check if it works as expected.

Reply:
I ran your tests removing the is healthy logic and they all passed, and the tests look good, so I think this is Ok, but give it a peruse anyway. If you think I'm right, this means update a pair of comments in the tests that mention this logic. If I'm not, I think the `peers_removed += 1;` deserves it's own comment explaining logic for the future. Otherwise I think your PR is good to go

Reply:
Okay will take a look! That'd be good if we can remove the condition since yea it is quite confusing.

Reply:
I think you're right, the "is healthy logic" is unnecessary. Like you said, the ```worst_connected_peers``` only has peers that are still connected. Thanks for the catch! I'll update to remove references to that logic.

Reply:
We can then use a while loop with ```disconnecting_peers.len()``` that I think you / Age suggested earlier.